metricbeat/module/elasticsearch/shard/data_xpack.go

```diff
@@ -104,3 +106,24 @@ func getSourceNode(nodeID string, stateData *stateStruct) (common.MapStr, error)
 		"name": nodeInfo.Name,
 	}, nil
 }
+
+func getEventID(stateID string, shard common.MapStr) string {
+	var nodeID string
+	if shard["node"] == nil {
+		nodeID = "_na"
```

Comment:
Is that the default in monitoring? Interesting :-)

Reply:
Indeed :)

https://github.com/elastic/elasticsearch/blob/ef5007b6d89fad848e8c0a706fdfda78f954c459/x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/collector/shards/ShardMonitoringDoc.java#L71